facebook/delphiFacebook/R/contingency_aggregate.R

```diff
@@ -0,0 +1,361 @@
+## Functions for performing the aggregations in an efficient way.
+
+#' Produce aggregates for all desired aggregations.
+#'
+#' Writes the outputs directly to CSVs in the directory specified by `params`.
+#' Produces output using all available data between `params$start_date` and 
+#' `params$end_date`, inclusive.
+#'
+#' @param df Data frame of individual response data.
+#' @param aggregations Data frame with columns `name`, `var_weight`, `metric`,
+#'   `group_by`, `compute_fn`, `post_fn`. Each row represents one aggregate
+#'   to report. `name` is the aggregate's base column name; `var_weight` is the 
+#'   column to use for its weights; `metric` is the column of `df` containing the
+#'   response value. `group_by` is a list of variables used to perform the 
+#'   aggregations over. `compute_fn` is the function that computes
+#'   the aggregate response given many rows of data. `post_fn` is applied to the
+#'   aggregate data after megacounty aggregation, and can perform any final
+#'   calculations necessary.
+#' @param cw_list Named list of geographic crosswalks, each of which maps a zip5
+#'   to a geographic level such as county or state. Aggregates will be produced
+#'   for each geographic level.
+#' @param params Named list of configuration parameters.
+#'
+#' @return none
+#'
+#' @import data.table
+#' @importFrom dplyr full_join %>%
+#' @importFrom purrr reduce
+#'
+#' @export
+aggregate_aggs <- function(df, aggregations, cw_list, params) {
```

Comment:
any reason not to name this something a little less reduplicative? eg
```suggestion
produce_aggregates <- function(df, aggregations, cw_list, params) {
```